src/main/kotlin/org/rust/cargo/project/workspace/impl/CargoProjectWorkspaceImpl.kt

```diff
@@ -61,19 +62,27 @@ class CargoProjectWorkspaceImpl(private val module: Module) : CargoProjectWorksp
      */
     private var cached: CargoProjectDescription? = null
 
+    /**
+     * Isolated message-bus to insulate cargo-project-workspace related messaging
+     * infra from general-purpose listeners, and properly manage listeners' lifetimes
+     */
+    private var messageBus: MessageBus? = null
+
     /** Component hooks */
 
     override fun getComponentName(): String = "org.rust.cargo.CargoProjectWorkspace"
 
     override fun initComponent() {
+        messageBus = MessageBusFactory.newMessageBus(CargoProjectWorkspace::class.java.name, module.messageBus)
```

Comment:
This message bus provides the same messages as `module.messageBus` (you still publish updates to the `module.messageBus` and not to this private one). is this intentional?